src/python/pants/option/global_options.py

```diff
@@ -1718,6 +1718,19 @@ class GlobalOptions(BootstrapOptions, Subsystem):
         default=[],
     )
 
+    work_dir = StrOption(
```

Comment:
At a glance it's not obvious why we need this option. It looks like it's to communicate the cwd to the daemon, but it's not something the user will ever set, right? In which case could we just use a custom env var with some mangled name instead of an option? That way it won't appear in help output etc.